### PR TITLE
Moving to Jekyll

### PR DESCRIPTION
**Pull request by JustusAdam**
Modularizing is finished. Style still needs to be finished.

The old page can be found in the 'old' subdirectory.

Also added foundation css/js framework for style.

### CODE REVIEW

**Comment by heisig** [comment]
Ich hab mir das Ganze mal angeschaut. Meiner Meinung nach ist es etwas zu kompliziert gestaltet und im Moment noch in keiner Verfassung wo man es zusammenführen kann.

Ich würde darum bitten, dass wir uns wie in meiner Mail angesprochen möglichst bald treffen (füllt das Dudle aus!) um das weitere Vorgehen zu besprechen. Ich habe das Gefühl, dass wir verschiedene Meinungen zum Thema kleine, statische Webseite haben.

**Comment by MarauderXtreme** [comment]
In my opinion a website that is written with plain HTML/CSS/JS is okay if there aren't any changes for let's say months or years.
But we need this site to upload data, provide news and so on. For that reason it would be horrible to manually add every site again and copy all head/header/footer.
So at least as approach where something can be included is way better than this.

@heisig I filled out the dudle, so now we can search for a date.

**Comment by heisig** [comment]
Ich verstehe deinen/euren Standpunkt. Wobei bedacht werden muss, dass die Aktualisierungen der Seite sich in Grenzen halten. In der Woche soll der "News"-Kasten ein paar mal aktualisiert werden. Außerdem müssten die Bilder noch irgendwo Platz finden. Ansonsten bleiben die Inhalte statisch (und nach der ESE auch für Jahre).

Vielleicht müsst ihr mir euer Konzept auch einfach nur genauer erklären (das können wir dann bei einem Treffen ja machen). Im Moment bin ich mir z.B. nicht sicher warum noch Foundation eingebaut wurde bzw. eingebaut werden muss. Ich hatte erwartet, dass es ausreicht die aktuelle (alte) Webseite zu nutzen um ein Template zu erstellen. Ein mobiles Layout kann ich auch so erstellen indem ich mit CSS arbeite.

@MarauderXtreme Ich nehme mal an, dass du dann "Operator" im Dudle bist?!

**Comment by kiliankoe** [comment]
The ESE site is updated once (excl. additional changes for small updates) per year. As much as I like Jekyll and the options it provides, it really is somewhat overkill, at least in this form here.

New students will not be subscribing to any news information on this page. All they'll need for updates throughout the one week where they'll sporadically check the page is a highlighted header for any kind of updates if there should be any. I can't remember having updates for anything besides changes to their schedules, which will probably have to be published as plain static html anyways seeing how we're getting that data (and when...). So anything fancy for specific news posts is... yeah... not really necessary imho. The ESE page is not a blog, it's just static information that (in a best case scenario) shouldn't change at all once the page is online.

Things the current page does badly that should really be addressed are:
 - clean markup (especially regarding the week timetable) that can be edited within seconds by pretty much anyone responsible
 - responsive design as most people will be viewing the page from mobile
 - extremely easy ways of adjusting the layout and color for future generations (as far as this is possible)

If this can be solved through an easy jekyll setup, then so be it, but I'm currently not seeing the big plus here. This setup here doesn't address these three issues at all unfortunately.

**Comment by MarauderXtreme** [comment]
> * clean markup (especially regarding the week timetable) that can be edited within seconds by pretty much anyone responsible
> * responsive design as most people will be viewing the page from mobile
> * extremely easy ways of adjusting the layout and color for future generations (as far as this is possible)

These are the things that should be achieved by using foundation.
@heisig If you can do a mobile layout of this site with only writing own css that looks neat I would be delighted to take lessons in CSS from you. For me it is a hard task to get a structure to a mobile layout if this structure hasn't been designed mobile first.

I see your point that with only one site (index.html) it is not that important to keep the structure clean.
Using Jekyll mostly helps in creating content. Writing HTML is quite a headache task because you can easlily screw things up. Writing content in an easier markup language like Markdown, MediaWiki or whatever speeds things up and can prevent render fails.

**Comment by JustusAdam** [comment]
### First of all ...

\<rant severity="mild" friendliness=":arrow_up:" manners=false> :anger: 

> kleine, statische Webseite haben.

We never said small, only static.

> The ESE page is not a blog

Yes, I know and jekyll is no blogging software. :x: It's a blog-aware static site generator. :white_check_mark: 

> it's just static information that (in a best case scenario) shouldn't change at all once the page is online.

really? You should know by now that no software is ever finished. :disappointed: 

> extremely easy ways of adjusting the layout ...

hence jekyll (modular temlpates for the pages), this is, in my humble opinion, by far the greatest benefit to gain from this system. :gem:

> ... and color for future generations

hence sass (I envision in the future having to change only a single sass color variable to readjust the entire :page_facing_up: for a new semester). Though it always leaves open the option to change more.

> shouldn't change at all once the page is online

I agree, that's why I argued against Wordpress and for a static site in the first place. Remember? If it weren't for me you'd all still be wysiwyg'ing that unmaintainable heap of :hankey:.

> it really is somewhat overkill

Is it though? It's just to keep the sources clean. The page itself is pure and old school html.

> Ein mobiles Layout kann ich auch so erstellen indem ich mit CSS arbeite.

For every :one: of the at least 6 pages individually? Using your own :hand: written css? And you will take care of testing it? On every :iphone:? On every screensize? 

> I would be delighted to take lessons in CSS from you

Have you had a single :eyes: at the css for the old :page_facing_up:? No offence. Although ... maybe some offence :angry:.  It really was :hankey:. I have never been :scream:'ed at harder by my linter.

\</rant> :sunny: 

### Conclusion

I'd also like to apologize for the slight shock a pull request with 400 :page_facing_up:es changed (added) may have caused. It is actually far less, But I didn't bother to delete the vendor code at first.

I actually didn't want to respond to this at all yet, but I could not restrain myself, sorry.

Also  I don't thing this is the proper venue to discuss this, though it is very nice to be able to read through previous responses, a personal meeting would be a much better option I think.

:one: last thing: The structural setup for the :page_facing_up:s was dead easy and is (I believe :thought_balloon:) done. The only thing missing is adjusting the classes for the style (combining foundation **and** the old style provide far harder than anticipated), perhaps this might be a :sunny: time to be thinking about a more modern style altogether, perhaps moving away from the same style we've had **since '10**.

Well, this went on longer than expected. A result of how much I like to write markdown coupled with the ranting endurance only a nice :beer: can provide. 

... I should add some emoticons to relieve the tension ...

TL;DR: have an :octocat:

**Comment by sjmielke** [comment]
Zum Thema:



±1. Für alle. Oder so. Sagt mir Bescheid, wenns dafür auch ein cooles Github-Emoji gibt.

Ne, ernsthaft, das führt hier zu nix, da isses sinnvoller sich mal im Büro zusammenzusetzen und (bei zivilisierter Lauststärke) mal ein bisschen drüber zu reden, was das alles werden soll.

**Comment by kiliankoe** [comment]
dis gunna b gud

**Comment by JustusAdam** [comment]
I thought I was phrasing it mildly? Am I being too provocative? I'm not trying to start a fight (I think?)

**Comment by heisig** [comment]
Das Dudle hat gesprochen: Am Mittwoch den 22.07. um 13 Uhr treffen wir uns und können sprechen!

**Comment by MarauderXtreme** [comment]
Wait. What Dudle have you looked at? No one besides you has time there.

**Comment by heisig** [comment]
Upps, meinte auch den Tag davor.. Dienstag den 21.07. um 13 Uhr. Ne Mail schreibe ich euch auch noch mal
